common/containers/Tabs/SendTransaction/components/WalletInfo.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import translate, { translateRaw } from 'translations';
+import { IWallet } from 'libs/wallet';
+import { print } from 'components/PrintableWallet';
+import { Identicon, QRCode } from 'components/ui';
+import GenerateKeystoreModal from 'components/GenerateKeystoreModal';
+import './WalletInfo.scss';
+
+interface Props {
+  wallet: IWallet;
+}
+
+interface State {
+  address: string;
+  privateKey: string;
+  isPrivateKeyVisible: boolean;
+  isKeystoreModalOpen: boolean;
+}
+
+export default class WalletInfo extends React.Component<Props, State> {
+  public state = {
+    address: '',
+    privateKey: '',
+    isPrivateKeyVisible: false,
+    isKeystoreModalOpen: false
+  };
+
+  public componentDidMount() {
+    this.setWalletAsyncState(this.props.wallet);
+  }
+
+  public componentWillReceiveProps(nextProps) {
```

Comment:
Ideally this should be put into the `WalletDecrypt` component itself, I can get to that over the next day or so

Reply:
I could also see this being a simple HOC that just doesn't render until it gets these, and then passes them in as props. Let me know what you think the best route is.